File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int answer = 0;

        int l = 0, r = height.length-1;
        while(l < r) {
            int lH = height[l], rH = height[r];
            int vol;
            if(lH <= rH) {
                vol = (r-l)*lH;
                l++;
            }else {
                vol = (r-l)*rH;
                r--;
            }
            answer = Math.max(answer,vol);
        }
        
        return answer;
    }
}

File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        l, r = 0, len(height) - 1
        max_amount = 0
        while l < r:
            l_h, r_h = height[l], height[r]
            amount = (r - l) * min(l_h, r_h)
            max_amount = max(max_amount, amount)

            if l_h <= r_h:
                l += 1
            if l_h > r_h:
                r -= 1
        
        return max_amount